app/stagemanager.js

```diff
@@ -86,26 +86,35 @@ class StageManager extends MyTrello {
     }
 
     closeUnusedStages(data) {
-      var deferred = Q.defer();
-        var stages = data[0].map(value => value['name']),
-            self = this;
+      var deferred = Q.defer(),
+          stages = data[0].map(value => value['name']),
+          closing = [];
 
         data[1].forEach(function(trelloList) {
+          var listDefer = Q.defer();
+          closing.push(listDefer.promise);
             if (!(stages.includes(trelloList.name))) {
                 self.getListCards(trelloList.id)
                 .then(function(d) {
-                    self.closeList(d, trelloList.id).then(deferred.resolve);
-                })
-                .catch(deferred.reject);
-
+                    self.closeList(d, trelloList.id).then(listDefer.resolve);
```

Comment:
This is just a preference, but I've taken to unwinding promises so I don't have nesting (which is the problem promises are supposed to solve).  So I think personally, I might have written this like:

```javascript
.then(function(d) {
  return self.closeList(d, trelloList.id);
})
.then(deferred.resolve)
.catch(deferred.reject);
```